### PR TITLE
Add new Protocol Layer section to Build

### PR DESCRIPTION
**Pull request by mishmosh**
This section is for developers building tooling such as wallets, exchanges, and developer tools, who will need to interact directly with Filecoin’s protocol layer.

I pulled content from the document extensively researched & written by @longfeiWan9 @jenks-guo-filecoin, with some edits for grammar and clarity.

TODOs:
- There are some formatting bugs I still need to address.

Open questions:
- We should probably retitle & merge the current [NodeJS & Infura quickstart](https://docs.filecoin.io/build/get-started/) into this section. Thoughts?

### CODE REVIEW

**Comment by longfeiWan9** [comment]
@mishmosh I build and ran the code, and I can only see one file under `Protocol layer` side menu. 
image
I think the current code to render the docs could not read files under folder `build/protocol` other than `index.md` file.
Am I right on this? or there is some config we can do to make sure files under folder `build/protocol` are recognizable? @johnnymatthews

**Comment by johnnymatthews** [comment]
> I think the current code to render the docs could not read files under folder build/protocol other than index.md file.
Am I right on this? or there is some config we can do to make sure files under folder build/protocol are recognizable? @johnnymatthews -- @longfeiWan9 

I think it's because this file is set as `index.md`. I can fix that now.

File: content/en/build/protocol/building-on-protocol-layer.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+At the time of writing, the block height is 1956488. You can confirm the current block height via any block explorer, such as [Filfox](https://filfox.info/en).
+
+## Get tipset by height and find block CID
+
+Now we have the height, we can query the node to find the tipset by this height. A tipset is a set of blocks that each have the same height and parent tipset. Filecoin’s consensus mechanism allows zero or multiple blocks at each epoch; blocks from the same epoch are assembled into tipsets.
+
+Use `ChainGetTipSetByHeight` to get the tipset at a given block height:
+
+```
+curl -X POST 'https://api.node.glif.io' -H 'Content-Type: application/json' --data '{"jsonrpc":"2.0","id":1,"method":"Filecoin.ChainGetTipSetByHeight","params":[1918961]}' -H "Accept: application/json" -s | jq
```
Comment (longfeiWan9) [inline content/en/build/protocol/building-on-protocol-layer.md:65]: I test this command, it does not work as expected and throw an error `wrong param count (method 'Filecoin.ChainGetTipSetByHeight'): 1 != 2`。
I think it is related to the Glif node only maintaining the latest 2000 blocks. Maybe we need add a note here.

File: content/en/build/protocol/building-on-protocol-layer.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+You can see this block contains the miner’s address, the Ticket, the Proof of SpaceTime (PoSt), the CID of the parents where this block evolved from in the IPLD DAG, and its own CIDs. 
+
+A few definitions: 
+
+* **_Parents: _**CIDs of the parent tipset’s blocks
+* **_Miner: _**The actor address of the miner that produced this block
+* **_ElectionProof_**: Shows the ticket is a winning ticket
+* **_WinPoStProof_**: The WinningPoSt proof that the chosen miner must submit to the network 
+* **_Messages_: **CIDs of the messages packed into this block
```
Comment (longfeiWan9) [inline content/en/build/protocol/building-on-protocol-layer.md:187]:
```suggestion
* **_Parents_**: CIDs of the parent tipset’s blocks
* **_Miner_**: The actor address of the miner that produced this block
* **_ElectionProof_**: Shows the ticket is a winning ticket
* **_WinPoStProof_**: The WinningPoSt proof that the chosen miner must submit to the network 
* **_Messages_**: CIDs of the messages packed into this block
```

**Comment by johnnymatthews** [comment]
This PR has been superseded by https://github.com/filecoin-project/filecoin-docs/issues/1253 and https://github.com/filecoin-project/filecoin-docs/pull/1484 I believe. Gonna close, but please re-open if it turns out that I'm wrong (somewhat likely).

**Pull request closed without merging**